extensions/notebook/src/book/bookTreeItem.ts

```diff
@@ -137,4 +137,29 @@ export class BookTreeItem extends vscode.TreeItem {
 			return undefined;
 		}
 	}
+
+	/**
+	 * Helper method to find a child md file or Notebook with a specified URL
+	 * @param section The current section we're checking
+	 * @param url The url of the md file or Notebook we're searching for
+	 */
+	public findChildItem(url: string): any {
+		return this.findChildItemRecur(this, url);
+	}
+
+	private findChildItemRecur(section: any, url: string): any {
```

Comment:
Can we return something other than any here?

Reply:
Not really. We don't have typing information for Jupyter book TOC. We could make our own types but at the end of the day we still can't rely on the TOC JSON always being "valid" as we know it so all that would get us is dev-time typing and I didn't feel it was worth the effort to do that at this point. 

---
In reply to: [323449823](https://github.com/microsoft/azuredatastudio/pull/7155#discussion_r323449823) [](ancestors = 323449823)